fl4health/utils/losses.py

```diff
@@ -84,18 +204,20 @@ def clear(self) -> None:
         """
         raise NotImplementedError
 
-    @abstractmethod
     def compute(self) -> Losses:
         """
-        Compute aggregation of current list of losses if non-empty.
+        Compute average of current list of losses if non-empty.
 
         Returns:
-            Losses: New Losses object with aggregate of losses in losses_list.
-
-        Raises:
-            NotImplentedError: To be impemented by child class.
+            Losses: New Losses object with average of losses in losses_list.
         """
-        raise NotImplementedError
+        assert len(self.losses_list) > 0
+
+        # assume the type of the first loss object is the type of all loss objects
+        if isinstance(self.losses_list[0], EvaluationLosses):
```

Comment:
This is sort of a work around because we're allowing LossMeter to be "loosely" type in that it handles "Losses" objects. This might be more trouble than it's worth, but I feel like the "proper" thing for us to do is generically type the LossMeter with a typevar T that has a bound of Losses, then when we construct a `self.train_loss_meter = LossMeter.get_meter_by_type(loss_meter_type)`
and the validation analogue, we provide the type TrainingLosses and EvaluationLosses to each respectively.

Then this becomes just
```
T.aggregate(self)
```

There might be unforseen consequences of that. So let me know what you think?

Reply:
I think this is a great idea. Maybe the only thing is that we will probably need 2 loss meters, one for evaluation and one for training, for each client.

Reply:
If I remember correctly, I think basic client has 2 loss meters already 🤔.
```
self.train_loss_meter = LossMeter.get_meter_by_type(loss_meter_type)
self.val_loss_meter = LossMeter.get_meter_by_type(loss_meter_type)
```
So I'm hoping it sort of seamlessly drops in. However, definitely let me know if this creates a massive headache.

Reply:
I just found out it's impossible in python to do `T.aggregate(self)` because on runtime the real type of `T` is lost, there is currently an open issue about it:
https://github.com/python/typing/issues/629
I will need to store the type in the class so the solution will be a bit uglier but I still think it's worth it.

Reply:
Oh very interesting. We're on the cutting edge of generics usage 😂. I agree, storing the type is a little less fun, but hopefully not too much uglier.